build/i2c/src/lib.rs

```diff
@@ -1358,6 +1404,115 @@ impl ConfigGenerator {
         Ok(())
     }
 
+    fn declare_sensor_struct(
+        &mut self,
+        d: &I2cDevice,
+        struct_name: &str,
+    ) -> Result<()> {
+        // Manually unpack the field so that changes to the sensor types
+        // will require changes here as well.
```

Comment:
Hah! So maybe my above comment about destructuring to guard against future changes was not too nitpicky! :)

Reply:
Indeed, I am nothing if not pedantic!